symbolic-base/src/ZkFold/Symbolic/Data/Class.hs

```diff
@@ -24,15 +25,39 @@ import           ZkFold.Base.Data.Vector          (Vector)
 
 -- | A class for Symbolic data types.
 class SymbolicData x where
+
     type Context x :: (Type -> Type) -> Type
+    type Context x = GContext (G.Rep x)
+
     type Support x :: Type
+    type Support x = GSupport (G.Rep x)
+
     type Layout x :: Type -> Type
+    type Layout x = GLayout (G.Rep x)
 
     -- | Returns the circuit that makes up `x`.
     pieces :: x -> Support x -> Context x (Layout x)
+    default pieces
```

Comment:
The idea was to have a single _instance_ of `SymbolicData` like this: `instance (G.Generic x, GSymbolicData (G.Rep x)) => SymbolicData x` in order to not have users write it down at all, but this also works I guess

Reply:
Sorry, in light of #308, I think better solution would be to go your way with usual Generic deriving